File: python/maximum_depth_of_binary_tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def maxDepth(self, root):
        if root == None:
            depth = 0
        else:
            depth = max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1
        return depth

File: cpp/maximum_depth_of_binary_tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode *root) {
        int depth;
        if (!root)
            depth = 0;
        else
            depth = max(maxDepth(root->left), maxDepth(root->right)) + 1;
        return depth;
    }
};